pyrealm/demography/t_model_functions.py

```diff
@@ -145,6 +145,260 @@ def calculate_sapwood_masses(
+    r"""Calculate sapwood respiration.
+
+    Calculates the total sapwood respiration (:math:`R_{\cdot s}`) given the individual
+    sapwood mass (:math:`W_{\cdot s}`) and the sapwood respiration rate of the plant
+    functional type (:math:`r_{s}`) :cite:p:`{see Equation 13, }Li:2014bc`.
+
+    .. math::
+         R_{\cdot s} = W_{\cdot s} \, r_s
+
+    Args:
+        resp_s: The sapwood respiration rate
+        sapwood_mass: The individual sapwood mass
+    """
+    return sapwood_mass * resp_s
+
+
+def calculate_foliar_respiration(resp_f: Series, whole_crown_gpp: Series) -> Series:
+    r"""Calculate foliar respiration.
+
+    Calculates the total foliar respiration (:math:`R_{f}`) given the individual crown
+    GPP (:math:`P`) and the foliar respiration rate of the plant functional type
+    (:math:`r_{f}`). :cite:t:`Li:2014bc` remove foliar respiration as a constant
+    proportion of potential GPP before calculating GPP for the crown, but ``pyrealm``
+    treats this proportion as part of the definition of plant functional types.
+
+    .. math::
+         R_{f} = P \, r_f
+
+    Args:
+        resp_f: The foliar respiration rate
+        whole_crown_gpp: The individual whole crown GPP.
+    """
+    return whole_crown_gpp * resp_f
+
+
+def calculate_fine_root_respiration(
+    zeta: Series, sla: Series, resp_r: Series, foliage_mass: Series
+) -> Series:
+    r"""Calculate foliar respiration.
+
+    Calculates the total fine root respiration (:math:`R_{r}`) given the individual
+    foliage mass (:math:`W_f`), along with the fine root respiration rate (:math:`r_r`),
+    the ratio of fine root mass to foliage area (:math:`\zeta`) and the specific leaf
+    area (:math:`\sigma`) :cite:p:`{see Equation 13, }Li:2014bc`
+
+    .. math::
+         R_{r} = \zeta \sigma W_f r_r
+
+    Args:
+        zeta: The ratio of fine root mass to foliage area of the PFT.
+        sla: The specific leaf area of the PFT.
+        resp_r: The respiration rate of fine roots of the PFT.
+        foliage_mass: The individual foliage mass.
+    """
+
+    return zeta * sla * foliage_mass * resp_r
+
+
+def calculate_net_primary_productivity(
+    yld: Series,
+    whole_crown_gpp: Series,
+    foliar_respiration: Series,
+    fine_root_respiration: Series,
+    sapwood_respiration: Series,
+) -> Series:
+    r"""Calculate net primary productivity.
+
+    The net primary productivity (NPP, :math:`P_{net}`) is calculated as a plant
+    functional type specific yield proportion (:math:`y`) of the total GPP (:math:`P`)
+    for the individual minus respiration (:math:`R_m`), as the sum of the respiration
+    costs for foliage  (:math:`R_f`), fine roots  (:math:`R_r`) and sapwood
+    (:math:`R_s`).
+
+    .. math::
+        P_{net} = y (P - R_m) = y (P - W_{\cdot s} r_s - \zeta \sigma W_f r_r - P r_f)
+
+    Note that this differs from Equation 13 of :cite:t:`Li:2014bc`, which removes foliar
```

Comment:
Is there a citation for this approach?

Reply:
No - it's a "new" approach but only in the pretty trivial sense that the foliage respiration rate can be tuned.